dask/bytes/core.py

```diff
@@ -145,64 +149,63 @@ def read_bytes(urlpath, delimiter=None, not_zero=False, blocksize=2**27,
             raise TypeError("blocksize must be an integer")
         blocksize = int(blocksize)
 
-    blocks, lengths, machines = fs.get_block_locations(paths)
-    if blocks:
-        offsets = blocks
+    client = None
+    if hasattr(fs, 'get_block_locations'):
+        offsets, lengths, machines = fs.get_block_locations(paths)
+        try:
+            from distributed.client import default_client
+            client = default_client()
+        except (ImportError, ValueError):  # no distributed client
+            pass
```

Comment:
I would be comfortable deprecating this functionality.

Reply:
Agreed, if we are happy to accept arrow's hdfs (via java of libhdfs3) as equivalent or preferable to hdfs3. However, I think in general there may still be other file-system backends that have some kind of similar per-worker considerations, so I think there is some value in keeping this around if only as an example of that idea.

Reply:
I'd prefer to do this in a separate PR. I agree that this special case is probably rare, and it might be wise to rethink the implementation.